tests/upgrades-images-signed/upgrade_test.go

```diff
@@ -19,24 +19,48 @@ var _ = Describe("cOS Upgrade tests - Images signed", func() {
 	})
 	Context("After install", func() {
 		When("images are signed", func() {
-			It("upgrades to latest available (master) and reset", func() {
-				out, err := s.Command("cos-upgrade")
+			It("upgrades to a specific image and reset back to the installed version", func() {
+				out, err := s.Command("source /etc/os-release && echo $VERSION")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(out).ToNot(Equal(""))
+
+				version := out
+				By("upgrading to an old signed image")
+				out, err = s.Command("cos-upgrade --verify --docker-image raccos/releases-opensuse:cos-system-0.4.32")
 				Expect(err).ToNot(HaveOccurred())
 				Expect(out).Should(ContainSubstring("Upgrade done, now you might want to reboot"))
 				Expect(out).Should(ContainSubstring("Booting from: active.img"))
-				By("rebooting")
+
+				By("rebooting and checking out the version")
 				s.Reboot()
-				Expect(s.BootFrom()).To(Equal(sut.Active))
-			})
 
-			It("upgrades to a specific image and reset back to the installed version", func() {
+				out, err = s.Command("source /etc/os-release && echo $VERSION")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(out).ToNot(Equal(""))
+				Expect(out).ToNot(Equal(version))
+				Expect(out).To(Equal("0.4.32\n"))
+			})
+			// TODO(itxaka): Is this basically the same test as the one below??
+			It("fails to upgrade if verify is enabled on an unsigned version", func() {
+				// Using releases-amd64 as those images are not signed
+				out, err := s.Command("cos-upgrade --verify --docker-image raccos/releases-amd64:cos-system-0.4.18")
+				Expect(err).To(HaveOccurred())
+				Expect(out).Should(ContainSubstring("No valid trust data"))
+			})
+			It("fails to upgrade if verify is enabled on an unsigned upgrade channel", func() {
+				out, err := s.Command("sed -i 's|raccos/releases-.*|raccos/releases-amd64\"|g' /etc/luet/luet.yaml && cos-upgrade --verify")
+				Expect(out).Should(ContainSubstring("does not have trust data"))
+				Expect(err).To(HaveOccurred())
+			})
+			It("upgrades to an signed image with --verify and can reset back to the installed state", func() {
 				out, err := s.Command("source /etc/os-release && echo $VERSION")
 				Expect(err).ToNot(HaveOccurred())
 				Expect(out).ToNot(Equal(""))
 
 				version := out
-				By("upgrading to an old signed image")
-				out, err = s.Command("cos-upgrade --docker-image raccos/releases-opensuse:cos-system-0.5.0")
+
+				By("running cos-upgrade with --verify and an signed image")
+				out, err = s.Command("cos-upgrade --verify --docker-image raccos/releases-opensuse:cos-system-0.4.9-9")
```

Comment:
This test might not work, however this is not executed currently on the CI. So if tests pass, I would not mind about this, since there is #197 which is meant to solve these issues. It will popup there and conflicts can be easily solved there.

Reply:
argh, we gonna have a conflict here because as the verify args are changing around I moved some tests from signed <-> unsigned as they make more sense there